### PR TITLE
Add BlockEpilogue transaction to the API

### PR DESCRIPTION
**Pull request by igor-aptos**
## Description
We are surfacing newly added BlockEpilogue transaction into the API.

## Type of Change
- [x] Breaking change

## Which Components or Systems Does This Change Impact?
- [X] Full Node (API, Indexer, etc.)
- [X] Move/Aptos Virtual Machine

## How Has This Been Tested?
provided unit tests

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: protos/proto/aptos/transaction/v1/transaction.proto [hunk 1]
```diff
@@ -51,6 +51,7 @@ message Transaction {
     TRANSACTION_TYPE_STATE_CHECKPOINT = 3;
     TRANSACTION_TYPE_USER = 4;
     TRANSACTION_TYPE_VALIDATOR = 20;
+    TRANSACTION_TYPE_BLOCK_EPILOGUE = 23;
```
Comment (msmouse) [inline protos/proto/aptos/transaction/v1/transaction.proto:54]: This can actually be 5 or 21, right? :trollface:

Comment (msmouse) [reply protos/proto/aptos/transaction/v1/transaction.proto:54]: maybe 21 so it's always increasing at least :trollface:

Comment (igor-aptos) [reply protos/proto/aptos/transaction/v1/transaction.proto:54]: is 20 intentional or a mistake?

i.e. is there a reason txn_data below starts from 7 and doesn't overlap with this?

Comment (msmouse) [reply protos/proto/aptos/transaction/v1/transaction.proto:54]: I think it's a mistake, but @bowenyang007  @CapCap ?

Comment (igor-aptos) [reply protos/proto/aptos/transaction/v1/transaction.proto:54]: seems like original review for validator transactions was - dunno - https://github.com/aptos-labs/aptos-core/pull/11009/files/a453bea41db4e7f4a7f41d5cdc23257fc6d1e550#r1419756557

seems like txn_data below was deduping with other fields, not this enum, but that got mixed up - https://github.com/aptos-labs/aptos-core/pull/6269/files#diff-415e8460829f8a6e9e8c570675bd2c425e94b988a022f5914da912cb21141a2a


I'll update and make a comment

Comment (bowenyang007) [reply protos/proto/aptos/transaction/v1/transaction.proto]: It's fine either way. we just need these to be unique. 

@larry-aptos can you help test this?

**Review by gregnazario — COMMENTED** [review]
I think there needs to be Rosetta change for this

**Comment by igor-aptos** [comment]
@gregnazario  - like this - https://github.com/aptos-labs/aptos-core/pull/13471/commits/764b09ee250a0693c8312f779cc0f784f4ce0f58 ? anything else to change/test? i.e. SDK, etc?

**Review by msmouse — APPROVED** [review]

**Review by gregnazario — APPROVED** [review]

**Pull request merged**